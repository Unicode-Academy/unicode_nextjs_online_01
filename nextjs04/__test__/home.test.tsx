import Home from "@/app/page";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

describe("HomePage", () => {
  it("render title", () => {
    render(<Home />);
    const title = screen.getByText("HomePage");
    expect(title).toBeInTheDocument();
  });
  it("renders a heading", () => {
    render(<Home />);
    const heading = screen.getByRole("heading", {
      level: 2,
      name: /nextjs/i,
    });
    expect(heading).toBeInTheDocument();
  });
  it("renders a button", () => {
    render(<Home />);
    const button = screen.getByRole("button", {
      name: "Click me",
    });
    expect(button).toBeInTheDocument();
  });
  it("renders homepage unchanged", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
