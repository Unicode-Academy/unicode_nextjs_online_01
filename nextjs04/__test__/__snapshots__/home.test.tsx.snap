// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders homepage unchanged 1`] = `
<div>
  <div>
    <h1>
      HomePage
    </h1>
    <h2>
      Chào mừng bạn đến với nextjs
    </h2>
    <h3>
      Hello NextJS
    </h3>
    <button>
      Click me
    </button>
    <h4>
      Hoàng An
    </h4>
  </div>
</div>
`;
